controller/api/public/stat_summary.go

```diff
@@ -0,0 +1,334 @@
+		labels[string(k)] = string(v)
+	}
+	return labels
+}
+
+func (h *handler) getDeployment(namespace string, name string) ([]appsv1.Deployment, map[string]*meshedCount, error) {
+	if namespace == "" {
+		namespace = apiv1.NamespaceDefault
+	}
+
+	deploymentsClient := h.k8sClient.Apps().Deployments(namespace)
+	deployment, err := deploymentsClient.Get(name, metav1.GetOptions{})
+	if err != nil {
+		return nil, nil, err
+	}
+
+	meshCount, err := h.getMeshedPodCount(namespace, deployment)
+	if err != nil {
+		return nil, nil, err
+	}
+	meshMap := map[string]*meshedCount{deployment.Name: meshCount}
+	return []appsv1.Deployment{*deployment}, meshMap, nil
+}
+
+func (h *handler) getDeployments(namespace string) ([]appsv1.Deployment, map[string]*meshedCount, error) {
+	if namespace == "" {
+		namespace = apiv1.NamespaceDefault
+	}
+
+	deploymentsClient := h.k8sClient.Apps().Deployments(namespace)
+	list, err := deploymentsClient.List(metav1.ListOptions{})
+	if err != nil {
+		return nil, nil, err
+	}
+
+	meshedPodCount := make(map[string]*meshedCount)
+	for _, item := range list.Items {
+		// TODO: parallelize
+		meshCount, err := h.getMeshedPodCount(namespace, &item)
+		if err != nil {
+			return nil, nil, err
+		}
+		meshedPodCount[item.Name] = meshCount
+	}
+
+	return list.Items, meshedPodCount, nil
+}
+
+// this takes a long time for namespaces with many pods
+func (h *handler) getMeshedPodCount(namespace string, obj runtime.Object) (*meshedCount, error) {
+	selector, err := getSelectorFromObject(obj)
+	if err != nil {
+		return nil, err
+	}
+
+	pods, err := h.getAssociatedPods(namespace, selector)
+	if err != nil {
+		return nil, err
+	}
+
+	meshCount := &meshedCount{}
+	for _, pod := range pods {
+		if isInMesh(pod) {
+			meshCount.inMesh++
+		} else {
+			meshCount.notInMesh++
+		}
+	}
+
+	return meshCount, nil
+}
+
+func isInMesh(pod apiv1.Pod) bool {
+	_, ok := pod.Annotations[k8s.ProxyVersionAnnotation]
+	return ok
+}
+
+func (h *handler) getAssociatedPods(namespace string, selector map[string]string) ([]apiv1.Pod, error) {
+	var podList *apiv1.PodList
+
+	selectorSet := labels.Set(selector).AsSelector().String()
+	if selectorSet == "" {
+		return podList.Items, nil
+	}
+
+	podsClient := h.k8sClient.Core().Pods(namespace)
+	podList, err := podsClient.List(metav1.ListOptions{
+		FieldSelector: fields.Everything().String(),
+		LabelSelector: selectorSet,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return podList.Items, nil
+}
+
+func getSelectorFromObject(obj runtime.Object) (map[string]string, error) {
+	switch typed := obj.(type) {
+	case *appsv1.Deployment:
+		return typed.Spec.Selector.MatchLabels, nil
+
+	default:
+		return nil, fmt.Errorf("Cannot get object selector: %v", obj)
+	}
+}
+
+func (h *handler) queryProm(ctx context.Context, query string) (model.Vector, error) {
+	log.Debugf("Query request: %+v", query)
+	end := time.Now()
+
+	// single data point (aka summary) query
+	res, err := h.prometheusAPI.Query(ctx, query, end)
```

Comment:
you can set time to zero and it will imply get the latest value. this is a bug that was recently fixed in the prometheus client.
```go
res, err := h.prometheusAPI.Query(ctx, query, time.Time{})
```

Reply:
oh nice!